zipkin/src/main/java/zipkin/internal/ThriftCodec.java

```diff
@@ -110,6 +110,7 @@ public Span readSpan(byte[] bytes) {
     final Field IPV4 = new Field(TYPE_I32, 1);
     final Field PORT = new Field(TYPE_I16, 2);
     final Field SERVICE_NAME = new Field(TYPE_STRING, 3);
+    final Field IPV6 = new Field(TYPE_STRING, 4);
```

Comment:
ah I think I understand why you asked! TType.STRING is used (because there's no TType.BINARY). the binary keyword in IDL is a hint, but the actual thrift type is string.

Ex. thrift makes this out of `4: optional binary ipv6`

```java
  private static final org.apache.thrift.protocol.TField IPV6_FIELD_DESC = new org.apache.thrift.protocol.TField("ipv6", org.apache.thrift.protocol.TType.STRING, (short)4);
```